polars/11_missing_data.py

```diff
@@ -0,0 +1,190 @@
+# /// script
+# requires-python = ">=3.12"
+# dependencies = [
+#     "marimo",
+#     "polars==1.22.0",
+# ]
+# ///
+
+import marimo
+
+__generated_with = "0.11.13"
+app = marimo.App(width="medium")
+
+
+@app.cell
+def _():
+    import marimo as mo
+    return (mo,)
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        """
+        # Handling Missing Data in Polars
+
+        _By Felix Najera._  
+
+        In real‑world datasets, missing values are inevitable. Polars offers a rich set of methods to detect, remove, and impute missing data efficiently. This notebook walks through common patterns and best practices for dealing with nulls in Polars DataFrames.
+        """
+    )
+    return
+
+
+@app.cell
+def _(mo):
+    mo.accordion({
+        "### TLDR":"""You can identify and manage missing values using specific methods: [```.is_null```](https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.is_null.html) to check for missing values, [```fill_null```](https://docs.pola.rs/api/python/stable/reference/expressions/api/polars.Expr.fill_null.html) replaces them with specified values, and [`drop`](https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.drop.html) removes rows or columns containing null values, applicable to all data types including strings. Taking special care of variables with String cases see the Strings notebook for more."""
+
+    })
+    return
+
+
+@app.cell
+def _(mo):
+    mo.md("""Polars datatype specific features for missing data are NaN(for float values) and null(for everything else).""")
+    return
+
+
+@app.cell
+def _():
+    import polars as pl
+
+    df = pl.DataFrame(
+        {
+            "A": [1, 2, 3, 4, 5],
+            "fruits": ["banana", None, "apple", "apple", "banana"],
+            "age": [25, None, 37, 29, None],
+            "B": [5, 4, 3, 2, 1],
+            "score": [85, 92,  None, None, 88],
+            "height_cm": [170.0, 165.5, None, 180.2, 175.0],
+
+        }
+    )
+    df
+    return df, pl
+
+
+@app.cell
+def _(mo):
+    mo.md("""Where ```null``` is what is used in dataset features of with datatype other floats and ```NaN``` is used in the dataset features witha  datatype of float""")
```

Comment:
```null``` may be used to indicate missing data in any data types. NaN might be used as a placeholder for missing values in some datasets with floats, specially in formats that lack proper support for null values, but it is not missing per se (the same way you may see some datasets give 0 or -1 a distinct meaning)